addons/calendar/models/calendar.py

```diff
@@ -1371,7 +1371,7 @@ def detach_recurring_event(self, values=None):
                 rrule_type=False,
                 rrule='',
                 recurrency=False,
-                final_date=datetime.strptime(data.get('start'), DEFAULT_SERVER_DATETIME_FORMAT if data['allday'] else DEFAULT_SERVER_DATETIME_FORMAT) + timedelta(hours=values.get('duration', False) or data.get('duration'))
+                final_date=False
```

Comment:
Ca peut surement marché aussi
sinon l'autre solution est de fixé le client web non ?

Reply:
in v7~saas-3, in a old old other world, final_date always contains the last end_date of an event.

So when you want search for event that start after 1/11 and stop before 1/12, you need to check on final date because a recurring event can have a stop before the 1/11 but a final_date after the 1/12.

Python2/python3/Orm (real date vs String)  automatically (at least in the past) trunc the datetime part.
And web Client never crash for it since 2014 before now (at first view)

So yes, I think that:
   if final_Date has always the same meaning that 5 versions ago, and if orm/python don't trunc datetime now, and if webclient crash now and not before: the fix will be probably to keep the final_date but cast it correctly in date +1day, and maybe see why webclient suddenly don't support anymore this field.
  if one of this 'if' is not true, ... do what seems the best for you :)